docs/topics/streamfield.md

```diff
@@ -575,6 +575,10 @@ hero_image = my_page.body.first_block_by_name('image')
 <div class="hero-image">{{ page.body.first_block_by_name.image }}</div>
 ```
 
+## Search considerations
+
+a `search_index` argument is added to block's constructor with `get_searchable_content` method so that if `search_index` is False it could return a empty list so that the given type of block should not be included in the search index.
```

Comment:
Again, this is talking too much about the internal implementation, rather than how to use it - the `get_searchable_content` method (and the way that it returns a list of strings) is an internal detail that we don't need to tell users about here.

(It might be useful to make `get_searchable_content` a properly documented function, to allow developers to do more advanced custom things with it... but if we were doing that, we'd need to go into more detail with example code - and I think that's more work than we want to do here.)

My suggested text:

> Like any other field, content in a StreamField can be made searchable by adding the field to the model's `search_fields` definition - see "Indexing extra fields". By default, all text content from the stream will be added to the search index. If you wish to exclude certain block types from being indexed, pass the keyword argument `search_index=False` as part of the block's definition. For example:

...followed by the example code snippet I originally gave in #11127.

("Indexing extra fields" should link to https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-extra-fields - I suggest referring to how that's done [on the intro paragraph for that page](https://github.com/wagtail/wagtail/blob/59b07f777f3468b5c4413c2e60a60af15587c5bb/docs/topics/search/indexing.md#L7), as the syntax for cross-linking in documentation isn't very obvious...)